partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -241,16 +242,95 @@ internal class PlanTyper(
-            TODO("Type RelOp Intersect")
+            val lhs = visitRel(node.lhs, node.lhs.type)
+            val rhs = visitRel(node.rhs, node.rhs.type)
+            // Check for Compatibility
+            if (!setOpSchemaSizesMatch(lhs, rhs)) {
+                return createRelErrForSetOpMismatchSizes()
+            }
+            if (!node.isOuter && !setOpSchemaTypesMatch(lhs, rhs)) {
+                return createRelErrForSetOpMismatchTypes()
+            }
+            // Compute Schema
+            val type = Rel.Type(lhs.type.schema, props = emptySet())
+            return Rel(type, node.copy(lhs = lhs, rhs = rhs))
         }
 
-        override fun visitRelOpExcept(node: Rel.Op.Except, ctx: Rel.Type?): Rel {
-            TODO("Type RelOp Except")
+        override fun visitRelOpUnion(node: Rel.Op.Union, ctx: Rel.Type?): Rel {
+            val lhs = visitRel(node.lhs, node.lhs.type)
+            val rhs = visitRel(node.rhs, node.rhs.type)
+            // Check for Compatibility
+            if (!setOpSchemaSizesMatch(lhs, rhs)) {
+                return createRelErrForSetOpMismatchSizes()
+            }
+            if (!node.isOuter && !setOpSchemaTypesMatch(lhs, rhs)) {
+                return createRelErrForSetOpMismatchTypes()
+            }
+            // Compute Schema
+            val size = max(lhs.type.schema.size, rhs.type.schema.size)
+            val schema = List(size) {
+                val lhsBinding = lhs.type.schema.getOrNull(it) ?: Rel.Binding("_$it", MISSING)
+                val rhsBinding = rhs.type.schema.getOrNull(it) ?: Rel.Binding("_$it", MISSING)
+                val bindingName = when (lhsBinding.name == rhsBinding.name) {
+                    true -> lhsBinding.name
+                    false -> "_$it"
+                }
+                Rel.Binding(bindingName, unionOf(lhsBinding.type, rhsBinding.type))
+            }
+            val type = Rel.Type(schema, props = emptySet())
+            return Rel(type, node.copy(lhs = lhs, rhs = rhs))
+        }
+
+        /**
+         * @return whether each type of the [lhs] is equal to its counterpart on the [rhs]
+         * @param lhs should be typed already
+         * @param rhs should be typed already
+         */
+        private fun setOpSchemaTypesMatch(lhs: Rel, rhs: Rel): Boolean {
+            // TODO: [RFC-0007](https://github.com/partiql/partiql-lang/blob/main/RFCs/0007-rfc-bag-operators.md)
+            //  states that the types must be "comparable". The below code ONLY makes sure that types need to be
+            //  the same. In the future, we need to add support for checking comparable types.
+            for (i in 0..lhs.type.schema.lastIndex) {
+                val lhsBindingType = lhs.type.schema[i].type
+                val rhsBindingType = rhs.type.schema[i].type
+                if (lhsBindingType != rhsBindingType) {
+                    return false
+                }
+            }
```

Comment:
Since the current schema type comparison is only partially implemented, I'll delete this validation block for the 0.14.6 release to unblock https://github.com/partiql/partiql-scribe/issues/55. The different Scribe dialects may have their own notion of comparability and perform their own set operator checks.